src/emm/services/modules_service.py

```diff
@@ -225,9 +230,113 @@ def git_commit_modules(self, commit: str) -> None:
 
         :param commit: Commit content for all changes.
         """
-        enabled_modules = self.get_all_modules(True)
         self.git_service.commit_all(commit)
+        enabled_modules = self.get_all_modules(True)
         for module in enabled_modules:
             module_data = self.get_module_data(module)
             module_location = Path(module_data.prefix) / module
             self.git_service.commit_all(commit, module_location)
+
+    def add_pr_links(self, comments: Dict[str, str]) -> Dict[str, List[str]]:
+        """
+        Add link to each module.
+
+        :param comments: Dictionary of module and its pull request url.
+        :return: Dictionary of pull request associate with other modules' pull requests.
+        """
+        pr_dict: Dict[str, List[str]] = {}
+        for module, pr_url in comments.items():
+            pr_links = "\n".join(
+                "{} pr: {}".format(repo, link) for repo, link in comments.items() if repo != module
+            )
+            pr_dict[module] = []
+            pr_dict[module].append(pr_url)
+            pr_dict[module].append(pr_links)
+        return pr_dict
+
+    def update_pr_links(self, comment_links: Dict[str, List[str]]) -> List[str]:
+        """
+        Update link to each module.
+
+        :param comment_links: Dictionary of module with its location and other modules' pull request link.
+        :return: List of pull requests being created associate with its link.
+        """
+        all_pull_requests = []
+        lines = (
+            "This code review is spread across multiple repositories. "
+            "Links to the other components of this review can be found:\n "
+        )
+        for module, pr_list in comment_links.items():
+            pr_url = pr_list[0]
+            if module != "base":
+                module_data = self.get_module_data(module)
+                module_location = Path(module_data.prefix) / module
+                pr_links = lines + pr_list[1]
+                self.github_service.pr_comment(pr_url, pr_links, module_location)
+                all_pull_requests.append(pr_url)
+                all_pull_requests.append(pr_links)
+            else:
+                pr_links = lines + pr_list[1]
+                self.github_service.pr_comment(pr_url, pr_links)
+                all_pull_requests.append(pr_url)
+                all_pull_requests.append(pr_links)
+
+        return all_pull_requests
+
+    def push_branch_to_remote(self, directory: Optional[Path] = None) -> None:
+        """
+        Determine if need push current branch to remote.
+
+        :param directory: Directory to execute the command at.
+        """
+        branch = self.git_service.current_branch(directory)
+        if branch == "main" or branch == "master":
```

Comment:
Could you pull "main" and "master" into a constant? It would probably be easiest to just make them a set and check if the branch is in the set:

```python
PROTECTED_BRANCHES = {"main", "master"}
...
if branch in PROTECTED_BRANCHES:
```